### ISSUE TITLE
Log in flow: trigger log in when »Stay logged in« checkbox is clicked

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/core#20477]
Currently the flow is:

1. fill in username
2. fill in password
3. (optional: check »Stay logged in« checkbox)
4. press enter or click on confirm icon

Now most of the time the »Stay logged in« checkbox will be checked after filling in the credentials, and when checking it to stay logged in you most likely will want to log in. So we should do that automatically:

When you check the »Stay logged in« checkbox to be checked, and _IF_ the username and password fields are filled out, basically trigger a click on the log in button.

cc @owncloud/designers @PVince81

**Comment by stefan-niedermann** [comment]
if username == filled and password == filled
text should be changed from "Stay logged in" to "Stay logged in and log in" or so, to make clear, that clicking this checkbox triggers the login action.
otherwise it's very confusing since a user does not expcect an direct action on clicking a checkbox.

**Comment by PVince81** [comment]
Clicking on a checkbox that triggers the login directly ? That will feel weird.
As a user I usually don't expect checkboxes to have such radical action.

**Comment by EtlamGit** [comment]
As this suggestion is based on some other stuff I reported, I would like to provide some clarification of my initial report here:
The new design has only that tiny login arrow inside the password box. Therefore in most cases I accidentally click on the "[ ] remember" field instead of the login button. jancborchardt made this suggestion to make some use of that accidental click.

I personally would also like to have the "[ ] remember" check box not to toggle + accepted if I accidentally click on it. Especially if the login is triggered automatically the state of this check box is evaluated, and possibly some cookies are generated on my PC. I prefer to have control if cookies are generated or not, especially if the PC is a shared one and I do the accidentally click.

**Comment by jancborchardt** [comment]
I made this suggestion not only because of your report @EtlamGit, but because checking the »Stay logged in« checkbox will most likely result in clicking the »Log in« arrow directly afterwards. This proposal just changes the flow and speeds it up.

@stefan-niedermann @PVince81 for reasons stated above I’d say that we should at least try it out.

> text should be changed from "Stay logged in" to "Stay logged in and log in" or so, to make clear, that clicking this checkbox triggers the login action.

@stefan-niedermann I don’t think that this is necessary at all and is just more text on an otherwise clean log in page.

**Comment by prastut** [comment]
I am GSOC 2016 aspirant and wish to take up this issue. How to proceed further:
* Do I have to sign the Contributor agreement?
* Correct me if the following workflow is wrong:
  * Fork core. Set remote to forked core
  * Open a branch from master. Any conventions to be followed?
  * Open a PR to master.

**Comment by MorrisJobke** [comment]
> Do I have to sign the Contributor agreement?

This would make it easier for you to contribute. You could also provide smaller pieces as MIT licensed code.

> Fork core. Set remote to forked core

As long as you didn't signed the CLA this is correct. Once signed you could create branches right inside the core repo.

> Open a branch from master. Any conventions to be followed?

Usually not. Best would be to have tests ready for the added feature, but this is sometimes hard to accomplish.

> Open a PR to master.

Correct. Then the review process will start automatically - see https://doc.owncloud.org/server/9.0/developer_manual/bugtracker/codereviews.html

**Comment by MorrisJobke** [comment]
@prastut And welcome aboard :)

**Comment by prastut** [comment]
@MorrisJobke Aye Aye Captain.

I wanted to discuss on the usability before actually working on this.  Issues I faced while brainstorming:
* In the initial thought process, I felt the users tend to think of unchecking or checking the check box (The user is in control perspective).
* If it's auto triggered, they lose their sense of control. What if I am just trying this feature out? 

Rather than auto trigger of login, some kind of visual cue would be best in terms of making the user think about his/her action. I get that we are trying to make the interaction more simpler, but at the cost of decreased userablitiy? Though  not having auto, means one more step with the user.
 In terms of visual cues: 
 1. The login arrow could become bigger and smaller once to tell user > you can proceed further.
 2. A jQuery slidedown of a div saying you can enter.

Both are self explanatory in that context, therefore I guess no need. I was wondering if anybody could elaborate more on this. The brainstorming on the issue isn't enough for me I felt.

**Comment by MorrisJobke** [comment]
@prastut What do you mean with auto trigger?

cc @owncloud/designers

**Comment by prastut** [comment]
@MorrisJobke As reported in the issue: 

When you check the »Stay logged in« checkbox to be checked, and IF the username and password fields are filled out, trigger a click on the log in button.

This is the auto trigger I am talking about.

**Comment by MorrisJobke** [comment]
> This is the auto trigger I am talking about.

Got it.

**Comment by jancborchardt** [comment]
@prastut as said on IRC this for sure is more experimental than fully thought through because well – no one else is doing this yet. I think it’s worth trying out. And by making it automatic, not adding even more stuff like strange visual cues which might not be well understood. This is a pure interaction enhancement.

TL;DR: As soon as you filled in your credentials, and you check »Stay logged in«, what are you going to do in the majority of cases? Probably log in.
Log in is something which people don’t want to bother with, so making this as seamless as possible is what we should aim for.

**Comment by prastut** [comment]
True. Will create a PR for this and let's see how it goes.

**Comment by MorrisJobke** [comment]
Fixed with #22815 in 9.1

**Comment by jancborchardt** [comment]
Great first contribution @prastut! :)

**Comment by prastut** [comment]
@jancborchardt Thanks for helping out too. :metal:
